tests/PHPStan/Analyser/data/loose-equal-consts.php

```diff
@@ -0,0 +1,92 @@
+<?php declare(strict_types=1);
+
+namespace LooseEqualConsts;
+
+use function PHPStan\Testing\assertType;
+
+class HelloWorld
+{
+	public function vars($mixed): void
+	{
+		if ($mixed === 'backend') {
+			assertType("'backend'", $mixed);
+		}
+
+		if ($mixed == 'backend') {
+			assertType("'backend'", $mixed);
+		}
+		if ($mixed == 500) {
+			assertType("500", $mixed);
+		}
+		if ('backend' == $mixed) {
+			assertType("'backend'", $mixed);
+		}
+		if (500 == $mixed) {
+			assertType("500", $mixed);
+		}
+		if ($mixed == 'backend' || $mixed == 'frontend') {
+			assertType("'backend'|'frontend'", $mixed);
+		}
+
+		if ($mixed == '1') {
+			assertType("mixed", $mixed);
+		}
+		if ($mixed == '0') {
+			assertType("mixed", $mixed);
+		}
+		if ($mixed == 1) {
+			assertType("mixed", $mixed);
+		}
+		if ($mixed == 0) {
+			assertType("mixed", $mixed);
+		}
```

Comment:
these 4 could be more precise

```suggestion
		if ($mixed == '1') {
			assertType("mixed~0|0.0|''|'0'|array{}|false|null", $mixed);
		}
		if ($mixed == '0') {
			assertType("0|0.0|''|'0'|array{}|false|null", $mixed);
		}
		if ($mixed == 1) {
			assertType("mixed~0|0.0|''|'0'|array{}|false|null", $mixed);
		}
		if ($mixed == 0) {
			assertType("0|0.0|''|'0'|array{}|false|null", $mixed);
		}
```

but I think its a unrelated problem for a followup?